### PR TITLE
Implmented route protection in UI

### LINKED ISSUE #380: [Task] Do not show the "Dataset Review" option on the Menu if not an admin
Description
At this moment, the "Dataset Review" is an option on the menu. This page should not be accessible or seen from anyone who is not an admin.

This requires verifying if the user is an admin.
 
This is part of the small tasks for the website.

### PR DESCRIPTION
**Pull request by ds161833**
what a not logged in user sees:
image
what a regular user sees:
image
What an admin sees:
image
If a user that is not logged in is trying to view a page they dont have permission to view they will see the message:
image

### CODE REVIEW

File: Client/src/Components/DatasetUpload/DatasetViewModal.tsx [hunk 1]
```diff
@@ -59,7 +59,6 @@ export const DatasetFormModal = (props: IProps) => {
             />
           </Box>
         </Paper>
-
```
Comment (laithAwad) [inline Client/src/Components/DatasetUpload/DatasetViewModal.tsx:62]: i think this is unneeded space

File: Client/src/Components/Navigation/ProtectedRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export const ProtectedRoute = (props: IProps) => {
+  const user = useUserSelector()
+  const { component, protectedRoute, exact } = { ...props }
+
+  const ProtectionMessage = () => {
+    return (
+      <Grid container direction="column" justify="center" alignItems="center">
+        <Grid item>
+          <Typography>You are not authorized to view this page</Typography>
```
Comment (laithAwad) [inline Client/src/Components/Navigation/ProtectedRoute.tsx:29]: Can we make this in the centre of the page? it is the main thing when the user is not signed in, and it should be very visible and large for the user

**Review by laithAwad — APPROVED** [review]
Great work, this will help me with the super admin story, so thank you

File: Client/src/Common/Consts/Routes.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  signUpRoute: ProtectedRouteModel,
+  forgotPasswordRoute: ProtectedRouteModel,
+  cellSizeAnalysisRoute: ProtectedRouteModel,
+  resetPasswordRoute: ProtectedRouteModel
+}
+
+export const routes: Routes = {
+  graphRoute: { route: '/graph/:graphStateId?', permission: 'anyone' },
+  newGraphRoute: { route: '/graph/', permission: 'anyone' },
+  homeRoute: { route: '/', permission: 'anyone' },
```
Comment (ratulkai93) [inline Client/src/Common/Consts/Routes.tsx:36]: this looks really nice and clean!

**Review by ratulkai93 — APPROVED** [review]
looks good overall. Thank you for the updates!

File: Client/src/Components/Navigation/ProtectedRoute.tsx (hunk 2 above)
Comment (ds161833) [reply Client/src/Components/Navigation/ProtectedRoute.tsx:29]: done!

**Pull request merged**